api/fma/v1alpha1/launcherpoolpolicy_types.go

```diff
@@ -0,0 +1,222 @@
+//     - For each unique combination of (Node, Accelerator, LauncherConfig), select the highest LauncherCount value
+//     - This forms the effective policy for that specific tuple
+//
+// 2. **Zero Matches**
+//   - When no LauncherPoolForNodeType matches a Node:
+//     - No pre-provisioning policy applies to that Node
+//     - Launcher pods for that Node are created on-demand
+//
+// ## Multiple CountForLauncher with Same LauncherConfig
+//
+// 1. **Duplicate LauncherConfig Names**
+//   - When multiple CountForLauncher structs specify the same LauncherConfigName for the same (Node, Accelerator) combination:
+//     - Select the highest LauncherCount value among all matching entries
+//     - This determines the target pool size for that LauncherConfig on that Node with that Accelerator
+//
+// 2. **Zero Matching CountForLauncher**
+//   - When no CountForLauncher matches a specific (Node, Accelerator, LauncherConfig) tuple:
+//     - No pre-provisioning occurs for that specific combination
+//     - Launcher pods are created on-demand when needed
+//
+// +genclient
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:shortName=lpp
+
+type LauncherPoolPolicy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   LauncherPoolPolicySpec   `json:"spec,omitempty"`
+	Status LauncherPoolPolicyStatus `json:"status,omitempty"`
+}
+
+// LauncherPoolPolicySpec defines the node-level idle pool configuration.
+type LauncherPoolPolicySpec struct {
+	// LauncherPoolForNodeType defines pool spec per node type.
+	LauncherPoolForNodeType []LauncherPoolForNodeType `json:"launcherPoolForNodeType"`
+}
+
+// LauncherPoolForNodeType defines launcher count for a class of nodes.
+type LauncherPoolForNodeType struct {
+	// Selector describes the hardware characteristics of target nodes.
+	//
+	// Introduce an EnhancedNodeSelector that supports combining label-based
+	// matching with resource field conditions.
+	// For example:
+	// enhancedNodeSelector:
+	//  # 1. Label selector (compatible with existing metav1.LabelSelector)
+	//  labelSelector:
+	//    matchLabels:
+	//      nvidia.com/gpu.family: ada-lovelace
+	//    matchExpressions:
+	//      - key: node.kubernetes.io/instance-type
+	//        operator: In
+	//        values: ["gx3-48x240x2l40s", "gx3-96x480x4l40s"]
+	//
+	//  # 2. Resource condition selector (new capability)
+	//  resourceRequirements:
+	//    allocatable:
+	//      cpu:
+	//        min: "16"
+	//        max: "64"
+	//      memory:
+	//        min: "128Gi"
+	//        max: "512Gi"
+	//      accelerators:
+	//        nvidia.com/gpu:
+	//          min: "2"
+	//	        max: "8"
+	//  acceleratorSelector:
+	//    type: "nvidia.com/gpu"
+	//    memory: "48Gi"
+	//    count: 4
+	// +required
+	EnhancedNodeSelector EnhancedNodeSelector `json:"enhancedNodeSelector"`
+
+	// PerAcceleratorCount defines pre-configuration quantities for each accelerator type
```

Comment:
It's actually per _homogeneous set_ of accelerators, right?

Reply:
yes